src/vaults/restaking/AffineReStaking.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.16;
+
+// upgrading contracts
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import {AccessControlUpgradeable} from "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import {PausableUpgradeable} from "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+
+// governance contract
+import {AffineGovernable} from "src/utils/audited/AffineGovernable.sol";
+
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {IWETH} from "src/interfaces/IWETH.sol";
+import {ReStakingErrors} from "src/libs/ReStakingErrors.sol";
+
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+
+contract AffineReStaking is UUPSUpgradeable, AccessControlUpgradeable, PausableUpgradeable, AffineGovernable {
```

Comment:
should we setup a function to migrate the assets to a new contract just in case?